### ISSUE TITLE
RotateMode.Fast parameter for Rigidbody2D extensions DORotate?

### ISSUE THREAD

**Issue by jmp909** [Demigiant/dotween#214]
Hi,

I noticed the Rigidbody2D extensions don't have the` RotateMode.Fast` parameter to calculate the shortest angle (they are in the Rigidbody 3D version)

would be great if these can be addedfor 2D

thanks
J

**Comment by Demigiant** [comment]
Ahoy,

For a Rigidbody2D rotation is just a float value change, not a full Quaternion like with 3D objects, so RotateMode doesn't make much sense there (under the hood, `Fast` means "the default Unity way for Quaternions, which calculates the shortest route", and there's no similar default for a single axis float rotation). Will think more about this though.

**Comment by jmp909** [comment]
I used the UnwrapAngle implementation for the start and end positions 

https://stackoverflow.com/a/20111513

“Simply unwrap the angle. Then you'll always have angles from 0 to 360. Unwrap the starting angle and the target angle, then perform your turn.”

As far as I can tell that then let’s Unity / DOTween take the shortest angle. But having that performed by DOTween internally would be useful 

I’d have to verify that’s the case though

Basically I’m adding a Random +/- 20 degrees to my start rotation , and setting the end rotation to the original rotation... so it rotates a small amount into the desired position. Without UnwrapAngle it was taking the long route sometimes

I guess a RotateMode.Clockwise and RotateMode.AntiClockwise would also be useful so you can force it.

**Comment by jmp909** [comment]
my object is kinematic. I'm actually finding that rotating the transform is a lot smoother. if i rotate the rigidbody the animation is stepped not continuous. any idea why this might be? is it because I am also tweening the transform?

```C#
transform.DOMoveY(targetY, tweenTime);

// this version is smooth
Vector3 targetRotation3D = new Vector3(0, 0, targetRotation);
transform.DORotate(targetRotation3D, tweenTime, RotateMode.Fast);

// this version isn't smooth between frames
Rigidbody2D rigidBody = b.RigidBody;
rigidBody.DORotate(targetRotation, tweenTime);
```
